### PR TITLE
Scraping

### PR DESCRIPTION
**Pull request by globaevg**

### CODE REVIEW

File: server/scrape/hatuScrape.js [hunk 1]
```diff
@@ -1,77 +1,99 @@
...
-const arrHaty = [];
+exports.hatuScrape = () => {
+  return new Promise((resolve, reject) => {
+    try {
+      function hScrape(i) {
+        if (isNaN(i)) {
+          i = 0;
+        }
+
+        request(
```
Comment (antonstan) [inline server/scrape/hatuScrape.js]: axios.get instead of request

File: server/scrape/index.js [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+
+const saveFlatsFromReserve = async () => {
+  try {
+    await FlatsReserve.collection.find().forEach(async rFlat => {
+      try {
+        const isOriginalFlatExist = await Flat.findOne({
+          fromUrl: rFlat.fromUrl
+        });
+        if (!isOriginalFlatExist) {
+          flat = new Flat(
```
Comment (antonstan) [inline server/scrape/index.js]: why do you not create using original object? new Flat(rFlat);

File: server/scrape/index.js [hunk 3]
```diff
@@ -0,0 +1,77 @@
...
+    });
+    return console.log('completed save flats');
+  } catch (error) {
+    console.log('erron inside of scrapeOfReb ' + error);
+  }
+};
+
+const deleteFlats = async () => {
+  try {
+    await Flat.collection.find().forEach(async oFlat => {
```
Comment (antonstan) [inline server/scrape/index.js]: to repositiries

File: server/scrape/index.js (hunk 2 above)
Comment (antonstan) [reply server/scrape/index.js]: and move it to the repositories

File: server/scrape/index.js [hunk 4]
```diff
@@ -0,0 +1,77 @@
...
+      if (oFlat.fromUrl) {
+        const isReserveFlatExist = await FlatsReserve.findOne({
+          fromUrl: oFlat.fromUrl
+        });
+        if (!isReserveFlatExist) {
+          Flat.collection.remove({ fromUrl: oFlat.fromUrl });
```
Comment (antonstan) [inline server/scrape/index.js]: to repo as well

File: server/scrape/index.js [hunk 5]
```diff
@@ -0,0 +1,77 @@
...
+
+    return console.log('delete flats is completed');
+  } catch (error) {
+    console.log('erron inside of deleteFlat ' + error);
+  }
+};
+
+const clearFlatsReserve = async () => {
+  try {
+    await FlatsReserve.deleteMany();
```
Comment (antonstan) [inline server/scrape/index.js]: =//=

File: server/scrape/rebScrape.js [hunk 6]
```diff
@@ -0,0 +1,140 @@
...
+      square_meters_number,
+      flore_number,
+      phone_number,
+      price_per_month,
+      pictures,
+      description,
+      incoming_date
+    };
+
+    console.log(obj.fromUrl);
```
Comment (antonstan) [inline server/scrape/rebScrape.js]: useless

File: server/scrape/test.js [hunk 7]
```diff
@@ -0,0 +1,75 @@
...
+// const { rebScrape } = require('./rebScrape');
+const Flat = require('../models/Flat');
+
+const deleteFlats = async () => {
+  try {
+    console.log('start delete Flats');
+    await Flat.collection.find(fromUrl).forEach(async oFlat => {
+      console.log(oFlat);
+    });
+    // await FlatsReserve.collection.find().forEach(async rFlat => {
```
Comment (antonstan) [inline server/scrape/test.js]: hmm

File: server/scrape/test.js [hunk 8]
```diff
@@ -0,0 +1,75 @@
...
+
+const two = () => {
+  return new Promise((resolve, reject) => {
+    setTimeout(() => {
+      resolve(console.log('two result'));
+    }, 2000);
+  });
+};
+
+const three = () => {
```
Comment (antonstan) [inline server/scrape/test.js]: wow, what is it ?

File: server/scrape/test.js [hunk 9]
```diff
@@ -0,0 +1,75 @@
+// const { hatuScrape } = require('./hatuScrape');
+// const { rebScrape } = require('./rebScrape');
+const Flat = require('../models/Flat');
+
+const deleteFlats = async () => {
+  try {
+    console.log('start delete Flats');
+    await Flat.collection.find(fromUrl).forEach(async oFlat => {
```
Comment (antonstan) [inline server/scrape/test.js]: repo

File: server/scrape/hatuScrape.js [hunk 10]
```diff
@@ -1,77 +1,99 @@
 const request = require('request');
 const cheerio = require('cheerio');
+const { saveReserveFlat } = require('../repositories/reserveFlats');
 
-const arrHaty = [];
+exports.hatuScrape = () => {
+  return new Promise((resolve, reject) => {
+    try {
+      function hScrape(i) {
```
Comment (antonstan) [inline server/scrape/hatuScrape.js]: ???

File: server/scrape/test.js (hunk 8 above)
Comment (globaevg) [reply server/scrape/test.js]: this is unnecessary file

File: server/scrape/test.js (hunk 9 above)
Comment (globaevg) [reply server/scrape/test.js]: this is unnecessary file too

File: server/scrape/hatuScrape.js (hunk 1 above)
Comment (globaevg) [reply server/scrape/hatuScrape.js]: completed

File: server/scrape/index.js (hunk 2 above)
Comment (globaevg) [reply server/scrape/index.js]: completed

File: server/scrape/index.js (hunk 3 above)
Comment (globaevg) [reply server/scrape/index.js]: completed

File: server/scrape/index.js (hunk 4 above)
Comment (globaevg) [reply server/scrape/index.js]: completed

File: server/scrape/index.js (hunk 5 above)
Comment (globaevg) [reply server/scrape/index.js]: completed

File: server/scrape/rebScrape.js (hunk 6 above)
Comment (globaevg) [reply server/scrape/rebScrape.js]: deleted

File: server/scrape/hatuScrape.js (hunk 10 above)
Comment (globaevg) [reply server/scrape/hatuScrape.js]: fixed

File: server/scrape/test.js (hunk 7 above)
Comment (globaevg) [reply server/scrape/test.js]: deleted file

Comment (globaevg) [reply server/scrape/test.js]: this file was unnessecary

File: server/scrape/hatuScrape.js [hunk 11]
```diff
@@ -1,24 +1,36 @@
-const request = require('request');
+const axios = require('axios');
 const cheerio = require('cheerio');
 
-const arrHaty = [];
+exports.hatuScrape = callback => {
+  return new Promise((resolve, reject) => {
+    try {
+      innerScrape = async (i = 0) => {
```
Comment (antonstan) [inline server/scrape/hatuScrape.js]: lets name variable "i" as "page" for more readable

File: server/scrape/hatuScrape.js [hunk 12]
```diff
@@ -1,24 +1,36 @@
...
+        const html = await axios.get(
```
Comment (antonstan) [inline server/scrape/hatuScrape.js]: name this variable as response and below us it as response.data

File: server/scrape/hatuScrape.js [hunk 13]
```diff
@@ -1,24 +1,36 @@
-const request = require('request');
+const axios = require('axios');
 const cheerio = require('cheerio');
 
-const arrHaty = [];
+exports.hatuScrape = callback => {
+  return new Promise((resolve, reject) => {
```
Comment (antonstan) [inline server/scrape/hatuScrape.js]: remove promise and make this function async

**Pull request merged**